### PR TITLE
Small refactoring of tests in CAN

### PR DESCRIPTION
**Pull request by strongly-typed**
I am happy to discuss the name `CanDriver`. `modm` just calls it [`modm::Can`](https://github.com/modm-io/modm/blob/develop/src/modm/architecture/interface/can.hpp#L25).

### CODE REVIEW

**Comment by strongly-typed** [comment]
This [bug](https://osdn.net/projects/mingw/ticket/39224) hit the Windows CI.

File: src/modm/architecture/interface/can_message.cpp.in [hunk 1]
```diff
@@ -10,22 +10,36 @@
...
+	if ((cc >= '0') and (cc <= '9')) {
+		return cc - '0';
+	} else if ((cc >= 'A') and (cc <= 'F')) {
+		return cc - 'A' + 10;
+	} else if ((cc >= 'a') and (cc <= 'f')) {
+		return cc - 'a' + 10;
+	} else {
+		return 0;
+	}
+}
```
Comment (salkinium) [inline src/modm/architecture/interface/can_message.cpp.in]: Technically `strtol(str, NULL, 16)` does this and also exists on avr-libc. However [looking at the implementation](https://github.com/vancegroup-mirrors/avr-libc/blob/master/avr-libc/libc/stdlib/strtol.c), it's fairly expensive.

Perhaps this code should be put into the `modm:math:algorithm` module?

Comment (strongly-typed) [reply src/modm/architecture/interface/can_message.cpp.in]: Yes, also tried strtol which seemed to be overkill. `algorithm` sound like QR decomposition and advanced stuff, not just stupid bit banging ...

There is `modm:math:utils:misc`🤓

I'll suggest `modm:math:utils:byte_operations`. There already is `modm:math:utils:bit_operations` which includes a `swap` that swaps the nibble of a byte.

**Review by salkinium — APPROVED** [review]
Thanks! We can deduplicate the hexchar code later, it's not that much overhead anyways.

**Comment by salkinium** [comment]
I changed the hexchar to a simple `uint64_t`, which is practically free.

**Pull request merged**